### PR TITLE
8 patch article by

### PR DESCRIPTION
**Pull request by ELR143**

### CODE REVIEW

File: models/news-model.js [hunk 1]
```diff
@@ -46,3 +46,14 @@ exports.selectCommentsByArticleId = (id) => {
...
+exports.updateArticleById = (incrementVotes, id) => {
+  const query = `UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *;`;
+  return db.query(query, [incrementVotes, id]).then((article) => {
+    if (article.rows[0].votes < 0) {
+      article.rows[0].votes = 0;
+      return article.rows[0];
+    }
+    return article.rows[0];
+  });
+};
```
Comment (nc-mick-fay) [inline models/news-model.js:86]: Great parameterisation of your query here!

**Review by nc-mick-fay — CHANGES_REQUESTED** [review]
Just missing one test for this endpoint:

- [x] Status 400: article_id is invalid (i.e. 'not-an-id')

Besides that everything looks great!

File: errors.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 exports.handleCustomErrors = (err, req, res, next) => {
-  res.status(400).send({ msg: "Error: 400 bad request" }).next(err);
+  if ((err.code = "22P02")) {
+    res.status(400).send({ msg: "Error: 400 bad request" }).next(err);
```
Comment (nc-jonathan-mcguire) [inline errors.js]: i dont think .next(err) does anything or is a method in this chain

File: errors.js [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+  } else if (err.status) {
```
Comment (nc-jonathan-mcguire) [inline errors.js]: if this is for your custom errors, id also check for err.msg -  alot of errors will have a status key, but only your custom errors will have .msg key

**Review by nc-jonathan-mcguire — APPROVED** [review]
Looks like the new test is there 👍

I've just left a comment about some possible refactoring you might want to do at some point

**Pull request merged**